src/labone/core/shf_vector_data.py

```diff
@@ -515,3 +598,92 @@ def parse_shf_vector_data_struct(
         return _deserialize_shf_waveform_vector(raw_data), None
     msg = f"Unsupported vector value type: {value_type}"
     raise ValueError(msg)
+
+
+def encode_shf_vector_data_struct(
+    data: np.ndarray | SHFDemodSample,
+    extra_header: ExtraHeader,
+) -> CapnpCapability:
+    """Encode the SHF vector data struct.
+
+    Build a capnp struct (in form of a dictionary) from data and
+    extra header to send.
+
+    Args:
+        data: The vector data.
+        extra_header: The extra header.
+
+    Returns:
+        The capnp struct to send.
+    """
+    # other versions not supported at the moment
+    version: _HeaderVersion = _HeaderVersion(major=0, minor=2)
+
+    if isinstance(extra_header, ShfScopeVectorExtraHeader):
+        if not isinstance(data, np.ndarray):
+            msg = "data must be of type np.ndarray for ShfScopeVectorExtraHeader"
+            raise TypeError(msg)
```

Comment:
nit-pick: What happens if you do not check this? I assume the code where you access the data fails ... could we just catch the exception coming from that access instead? This would be more pythonic since it follows the rule `better sorry than safe`

(also applies for the other header types)

Reply:
it may be that in the encryption itself no error occurs, but junk is send. Also, I think refactoring would lead to less readable code in this case. Especially since that an error occurs often only after the case distinction (if at all), making the construction of a useful error message even harder to read.